lightclient/src/background.rs

```diff
@@ -138,26 +154,28 @@ impl<TPlatform: PlatformRef, TChain> BackgroundTask<TPlatform, TChain> {
                 sender,
                 chain_id,
             } => {
-                let id = self.next_id(chain_id);
+                let (data, client) = self.for_chain_id(chain_id);
+                let id = data.next_id();
```

Comment:
Ok, we got chain_id -> ChainState now right should be good

I was afraid that we may re-use this globally but no worries.